pkg/tiller/release_update.go

```diff
@@ -69,8 +69,8 @@ func (s *ReleaseServer) prepareUpdate(req *services.UpdateReleaseRequest) (*rele
 		return nil, nil, errMissingChart
 	}
 
-	// finds the non-deleted release with the given name
-	currentRelease, err := s.env.Releases.Last(req.Name)
+	// finds the deployed release with the given name
+	currentRelease, err := s.env.Releases.Deployed(req.Name)
```

Comment:
If there is no "deployed" last release (ie: failed) then won't this prevent the code below from running? It seems the error handling should check for some "doesn't exist error" and if it doesn't exist, it should skip the `reuseValues` block, but otherwise continue.

Reply:
This is looking up what is live in the cluster to perform a diff of what needs to be installed/deleted/modified.  The code below is to calculate the incremented release number